coffee_plugin/src/plugin/plugin_mod.rs

```diff
@@ -70,104 +81,73 @@ fn from<T: Display>(err: T) -> PluginError {
     error!("{err}")
 }
 
-#[derive(Clone)]
-struct CoffeeInstall {
-    name: String,
-    usage: String,
-    description: String,
+#[rpc_method(
+    rpc_name = "coffee_install",
+    description = "install a plugin from one of the repository choosed"
+)]
+fn coffee_install(plugin: &mut Plugin<State>, request: Value) -> Result<Value, PluginError> {
+    let coffee = plugin.state.coffee();
+    let mut coffee = coffee.lock().unwrap();
+    let rt = Runtime::new().unwrap();
+
+    let request: InstallReq = serde_json::from_value(request)?;
+    rt.block_on(coffee.install(&request.name, false, true))
+        .map_err(from)?;
+    Ok(json!({}))
 }
 
-impl CoffeeInstall {
-    fn new() -> Self {
-        CoffeeInstall {
-            name: "coffee_install".to_string(),
-            usage: String::new(),
-            description: String::from("install a plugin from one of the repository choosed"),
-        }
-    }
-}
-
-impl RPCCommand<State> for CoffeeInstall {
-    fn call<'c>(
-        &self,
-        plugin: &mut Plugin<State>,
-        request: serde_json::Value,
-    ) -> Result<serde_json::Value, PluginError> {
-        let coffee = plugin.state.coffee();
-        let mut coffee = coffee.lock().unwrap();
-        let rt = Runtime::new().unwrap();
-
-        let request: InstallReq = serde_json::from_value(request)?;
-        rt.block_on(coffee.install(&request.name, false, true))
-            .map_err(from)?;
-        Ok(json!({}))
-    }
-}
-
-#[derive(Clone)]
-struct CoffeeList {
-    name: String,
-    usage: String,
-    description: String,
-}
-
-impl CoffeeList {
-    fn new() -> Self {
-        CoffeeList { name: "coffee_list".to_owned(), usage: String::new(), description: "show all the plugin installed and if {remotes} is specified show also the one available".to_owned() }
-    }
+#[rpc_method(
+    rpc_name = "coffee_list",
+    description = "show all the plugin installed and if {remotes} is specified show also the one available"
+)]
+fn coffee_list(plugin: &mut Plugin<State>, _: Value) -> Result<Value, PluginError> {
+    let runtime = Runtime::new().unwrap();
+    let coffee = plugin.state.coffee();
+    let mut coffee = coffee.lock().unwrap();
+    let result = runtime.block_on(coffee.list()).map_err(from)?;
+    Ok(serde_json::to_value(result)?)
 }
 
-impl RPCCommand<State> for CoffeeList {
-    fn call<'c>(
-        &self,
-        plugin: &mut Plugin<State>,
-        _: serde_json::Value,
-    ) -> Result<serde_json::Value, PluginError> {
-        let runtime = Runtime::new().unwrap();
-        let coffee = plugin.state.coffee();
-        let mut coffee = coffee.lock().unwrap();
-        let result = runtime.block_on(coffee.list()).map_err(from)?;
-        Ok(serde_json::to_value(result)?)
-    }
+#[rpc_method(rpc_name = "coffee_remote", description = "manage a remote")]
```

Comment:
This is giving a clippy warning 
```bash
warning: this `MutexGuard` is held across an `await` point
   --> coffee_plugin/src/plugin/plugin_mod.rs:111:1
    |
111 | #[rpc_method(rpc_name = "coffee_remote", description = "manage a remote")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
   --> coffee_plugin/src/plugin/plugin_mod.rs:111:1
    |
111 | #[rpc_method(rpc_name = "coffee_remote", description = "manage a remote")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock
    = note: `#[warn(clippy::await_holding_lock)]` on by default
    = note: this warning originates in the attribute macro `rpc_method` (in Nightly builds, run with -Z macro-backtrace for more info)

```

Reply:
We can doing nothing about it, it is a `warning: this `MutexGuard` is held across an `await` point` and it is better keep it to remember that we should change it at some point